projects/packages/password-checker/src/class-password-checker.php

```diff
@@ -0,0 +1,1313 @@
+					'required' => false,
+				),
+			),
+			'compare_to_list' => array(
+				'not_a_common_password'       => array(
+					'list_callback'    => 'get_common_passwords',
+					'compare_callback' => 'negative_in_array',
+					'error'            => __( 'This is a very common password. Choose something that will be harder for others to guess.', 'jetpack' ),
+					'required'         => true,
+				),
+				'not_same_as_other_user_data' => array(
+					'list_callback'    => 'get_other_user_data',
+					'compare_callback' => 'test_not_same_as_other_user_data',
+					'error'            => __( 'Your password is too weak: Looks like you are including easy to guess information about yourself. Try something a little more unique.', 'jetpack' ),
+					'required'         => true,
+				),
+			),
+		);
+
+		/**
+		 * Filters the password strength enforcement settings.
+		 *
+		 * You can determine the tests run and their order based on whatever criteria you wish to specify.
+		 *
+		 * @param array $tests tests to run.
+		 */
+		$tests = apply_filters( 'password_checker_tests', $tests );
+
+		if ( ! $sections ) {
+			return $tests;
+		}
+
+		$sections = (array) $sections;
+
+		return array_intersect_key( $tests, array_flip( $sections ) );
+	}
+
+	/**
+	 * Provides the regular expression tester functionality.
+	 *
+	 * @param string $password  the password.
+	 * @param array  $test_data the current test data.
+	 *
+	 * @return bool does the test pass?
+	 */
+	protected function test_preg_match( $password, $test_data ) {
+		return preg_match( $test_data['pattern'], $password );
+	}
+
+	/**
+	 * Provides the comparison tester functionality.
+	 *
+	 * @param string $password  the password.
+	 * @param array  $test_data the current test data.
+	 *
+	 * @return bool does the test pass?
+	 */
+	protected function test_compare_to_list( $password, $test_data ) {
+		if (
+			! is_callable( array( $this, $test_data['list_callback'] ) )
+			|| ! is_callable( array( $this, $test_data['compare_callback'] ) )
+		) {
+			return false;
+		}
+
+		return call_user_func(
+			array(
+				$this,
+				$test_data['compare_callback'],
+			),
+			$password,
+			call_user_func( array( $this, $test_data['list_callback'] ) )
+		);
+	}
+
+	/**
+	 * Getter for the common password list.
+	 *
+	 * @return array common passwords.
+	 */
+	protected function get_common_passwords() {
+		return $this->common_passwords;
+	}
+
+	/**
+	 * Returns the widely known user data that can not be used in the password to avoid
+	 * predictable strings.
+	 *
+	 * @return array user data.
+	 */
+	protected function get_other_user_data() {
+		if ( empty( $this->user_id ) ) {
+			return array();
+		}
+
+		$user_data = get_userdata( $this->user_id );
+		if ( ! $user_data ) {
+			return array();
+		}
+
+		if ( isset( $user_data->ID ) ) {
+			$this->add_user_strings_to_test( get_user_meta( $user_data->ID, 'first_name', true ) );
```

Comment:
@briancolinger for the compare_to_list tests. you can create a new user that has a first_name of "mysupercoolname" and then use that as the password to run the test.

Any of the data listed here can be written as tests (user_nicename, display_name, etc)

Reply:
Fixed.